orchestrator/orchestrator.go

```diff
@@ -90,7 +91,7 @@ func (s *Orchestrator) dependencyFilter(exec *execution.Execution) func(wf *proc
 		if len(parents) > 1 {
 			return false, fmt.Errorf("multi parents not supported")
 		}
-		return parents[0] == exec.NodeKey, nil
+		return parents[0] == exec.RefKey, nil
```

Comment:
`parents` are the list of parent nodes, so the list of `nodeKey`, I don't think this is right